## we will try to scrap the wikipaedia page containing the list of bollywood films

In [13]:
import requests
page = requests.get("https://en.wikipedia.org/wiki/Lists_of_Bollywood_films")

In [14]:
from bs4 import BeautifulSoup
soup = BeautifulSoup(page.content, 'html.parser')

In [15]:
print(soup.prettify())

<!DOCTYPE html>
<html class="client-nojs" dir="ltr" lang="en">
 <head>
  <meta charset="utf-8"/>
  <title>
   Lists of Bollywood films - Wikipedia
  </title>
  <script>
   document.documentElement.className=document.documentElement.className.replace(/(^|\s)client-nojs(\s|$)/,"$1client-js$2");RLCONF={"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":!1,"wgNamespaceNumber":0,"wgPageName":"Lists_of_Bollywood_films","wgTitle":"Lists of Bollywood films","wgCurRevisionId":903190497,"wgRevisionId":903190497,"wgArticleId":6200603,"wgIsArticle":!0,"wgIsRedirect":!1,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Articles with short description","Use dmy dates from March 2016","Use Indian English from March 2016","All Wikipedia articles written in Indian English","Articles needing additional references from October 2015","All articles needing additional references","Hindi-language films","Indian musical films","Lists of Bollywood films"],"wgBreakFrames":!1,"wgPageC

In [17]:
for url in soup.find_all('a'):
    print(url.get('href'))

None
#mw-head
#p-search
/wiki/File:Question_book-new.svg
/wiki/Wikipedia:Verifiability
//en.wikipedia.org/w/index.php?title=Lists_of_Bollywood_films&action=edit
/wiki/Help:Introduction_to_referencing_with_Wiki_Markup/1
//www.google.com/search?as_eq=wikipedia&q=%22Lists+of+Bollywood+films%22
//www.google.com/search?tbm=nws&q=%22Lists+of+Bollywood+films%22+-wikipedia
//www.google.com/search?&q=%22Lists+of+Bollywood+films%22+site:news.google.com/newspapers&source=newspapers
//www.google.com/search?tbs=bks:1&q=%22Lists+of+Bollywood+films%22+-wikipedia
//scholar.google.com/scholar?q=%22Lists+of+Bollywood+films%22
https://www.jstor.org/action/doBasicSearch?Query=%22Lists+of+Bollywood+films%22&acc=on&wc=on
/wiki/Help:Maintenance_template_removal
/wiki/Bollywood
/wiki/Mumbai
/wiki/Hindi_language
/wiki/Urdu
/wiki/Punjabi_language
#cite_note-1
#cite_note-2
/wiki/Indian_subcontinent
/wiki/Om_Shanti_Om
/wiki/Dhoom_2
/wiki/No_Entry
/wiki/Jodhaa_Akbar
/wiki/Fanaa_(film)
/wiki/Saawariya
/wiki/Kurbaan

## Now we will try to find the number of films produced in the last 100 years in bollywood.

In [73]:
import re
cnt= 0
film_list = []
for url in soup.findAll('a', href = re.compile("List_of_Bollywood_films")):
    actual_link = "https://en.wikipedia.org" + url.get('href')
    print(actual_link)
    cnt = cnt + 1
    film_list.append(actual_link)
    if cnt>100:
        break
        

https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2020
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2019
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2018
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2017
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2016
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2015
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2014
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2013
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2012
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2011
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2010
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2009
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2008
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2007
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2006
https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2005
https://

In [74]:
import requests
page_tab = requests.get("https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_1989")

soup_tab = BeautifulSoup(page_tab.content, 'html.parser')

In [75]:
tables = soup_tab.find_all('table')

In [76]:
print(tables[2].prettify())

<table class="wikitable">
 <tbody>
  <tr>
   <th>
    Title
   </th>
   <th>
    Director
   </th>
   <th>
    Cast
   </th>
   <th>
    Genre
   </th>
   <th>
    Music director
   </th>
  </tr>
  <tr>
   <td>
    <i>
     <a href="/wiki/Asmaan_Se_Ooncha" title="Asmaan Se Ooncha">
      Asmaan Se Ooncha
     </a>
    </i>
   </td>
   <td>
   </td>
   <td>
    <a href="/wiki/Jeetendra" title="Jeetendra">
     Jeetendra
    </a>
    ,
    <a href="/wiki/Sonam_(actress)" title="Sonam (actress)">
     Sonam
    </a>
   </td>
   <td>
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td>
    <i>
     <a href="/wiki/Awara_Zindagi" title="Awara Zindagi">
      Awara Zindagi
     </a>
    </i>
   </td>
   <td>
   </td>
   <td>
    <a class="mw-redirect" href="/wiki/Raj_Kiran_(Hindi_film_actor)" title="Raj Kiran (Hindi film actor)">
     Raj Kiran
    </a>
    ,
    <a href="/wiki/Jagdish_Raj" title="Jagdish Raj">
     Jagdish Raj
    </a>
   </td>
   <td>
   </td>
   <td>
   </td>
  </tr>
  <tr>
  

In [77]:
lis = tables[2].find_all('tr')
num = len(lis)
lis

[<tr>
 <th>Title</th>
 <th>Director</th>
 <th>Cast</th>
 <th>Genre</th>
 <th>Music director
 </th></tr>, <tr>
 <td><i><a href="/wiki/Asmaan_Se_Ooncha" title="Asmaan Se Ooncha">Asmaan Se Ooncha</a></i></td>
 <td></td>
 <td><a href="/wiki/Jeetendra" title="Jeetendra">Jeetendra</a>, <a href="/wiki/Sonam_(actress)" title="Sonam (actress)">Sonam</a></td>
 <td></td>
 <td>
 </td></tr>, <tr>
 <td><i><a href="/wiki/Awara_Zindagi" title="Awara Zindagi">Awara Zindagi</a></i></td>
 <td></td>
 <td><a class="mw-redirect" href="/wiki/Raj_Kiran_(Hindi_film_actor)" title="Raj Kiran (Hindi film actor)">Raj Kiran</a>, <a href="/wiki/Jagdish_Raj" title="Jagdish Raj">Jagdish Raj</a></td>
 <td></td>
 <td>
 </td></tr>, <tr>
 <td><i><a class="new" href="/w/index.php?title=Babul_(1989_film)&amp;action=edit&amp;redlink=1" title="Babul (1989 film) (page does not exist)">Babul</a></i></td>
 <td></td>
 <td></td>
 <td></td>
 <td>
 </td></tr>, <tr>
 <td><i><a href="/wiki/Bade_Ghar_Ki_Beti" title="Bade Ghar Ki Beti">

In [78]:
num

69

In [79]:
film_list

['https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2020',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2019',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2018',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2017',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2016',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2015',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2014',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2013',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2012',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2011',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2010',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2009',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2008',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2007',
 'https://en.wikipedia.org/wiki/List_of_Bollywood_films_of_2006',
 'https://

In [80]:
total = 0
for film in film_list:
    page_tab = requests.get(film)

    soup_tab = BeautifulSoup(page_tab.content, 'html.parser')
    tables = soup_tab.find_all('table')
    lis = tables[2].find_all('tr')
    num = len(lis)
    total = total + num
    
print(total)    

2674


### Hence the total number of bollywood films made in the last 100 years is 2674